pkg/ccl/backupccl/restore_job.go

```diff
@@ -448,6 +463,35 @@ func restore(
 		tasks = append(tasks, generativeCheckpointLoop)
 	}
 
+	procCompleteCh := make(chan struct{})
+	// countCompletedProcLoop is responsible for counting the number of completed
+	// processors. The goroutine returns a retryable error such that the job can
+	// replan if there are too many lagging nodes.
+	countCompletedProcLoop := func(ctx context.Context) error {
+		freq := replanFrequency.Get(&execCtx.ExecCfg().Settings.SV)
+		timer := time.NewTimer(freq)
+		defer timer.Stop()
+		done := ctx.Done()
+
+		for {
+			select {
+			case <-done:
+				return ctx.Err()
+			case _, ok := <-procCompleteCh:
+				if !ok {
+					return nil
+				}
+				timer.Reset(freq)
+			case <-timer.C:
```

Comment:
If you follow my other suggestion re timeutil.Timer, you'll want to set `timer.Read = true`.